### DISCUSSION TITLE
Is there a way to limit the On-Time minimum duration ?

### DISCUSSION THREAD

**Discussion by afaucogney** [jmcollin78/versatile_thermostat#220] [Q&A]
Hello,

  I have a heat-pump to warm my house that is built with compressors. Imagine, this is like a fridge system, with 2 circuits made by gas.
  Those compressor (I have 5 items) do not have to start and stop (flickering) all the time, to prevent its health.
  Is there a way to limit the minimun on-time duration to 30 min, the same for off-time.

**Comment by jmcollin78** [comment]
Hello yes. Search activation_duration_min (or something like that, I don't the readme under my eyes) in the readme.

**Comment by afaucogney** [comment]
Hello, so it seems this is not working. I have seen the param and setup to 30 min. 
You are talking about minimal_activation_delay ?
As I remember, it does limit the time between a reactivation (30 min between a stop and a new start), and I would expect it between a start and a stop !
So I need both in fact, that whatever the state, it does stay at least 30 min, and more if needed ! Does it theoricaly match with the timed algo ?

**Reply by jmcollin78** [reply]
Yes it is minimal_activation_delay. If you set it to 5 min, the cycle_min * on_percent should be > 5 min to activate the switch. Else the switch will stay off.

**Reply by afaucogney** [reply]
Thanks for the feedback.
I did the test and this was not working ! I had time "on" than "minimal_activation_delay" that was 30 min !
To be 100% to understand, because the percent work is confusing, when you says : 

`If you set it to 5 min, the cycle_min * on_percent should be > 5 min to activate the switch`

does it mean:
- it ALWAYS takes at least 5 min to activate the switch (a kind of latency)
- it is ALAWS at least 5 min in the state "on"
- ~ statically, is 5 min in the time "on"

**Reply by jmcollin78** [reply]
It means that VTherm will never turn on a heater for less that 5 min (the cycle in min * on_percent).

**Comment by MBRetrofit** [comment]
As I already explained in another discussion, the heat pump should heat a boiler with an automation based on anything but not the house heating. The Vtherm should drive the TRV's and the boiler, not the heating pump.
![f0cbf8_24ed8653d081456c9ce5a576dad9d008~mv2](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/814b6f4b-a6b4-4dd7-8cac-fd2c8e386c51)
